src/com/company/Customer.java

```diff
@@ -43,13 +44,25 @@ public Customer(UUID id) {
     }
 
     public void setUserName(String userName) {
-        this.userName = userName;
+
+        if (StaticConstants.CUSTOMER_LIST.stream().map(Customer::getUserName).anyMatch(s -> s.equals(userName))){
+
+            System.err.println("This user name already taken");
+
+            setEmail(new Scanner(System.in).nextLine());
```

Comment:
why we are calling setEmail method if the user name was already taken?

Reply:
good catch!